qemu/tests/virtio_chardev_trace.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    virtio-trace support testing:
+    1) Make FIFO per CPU in a host.
+    2) reboot guest with virtio-serial device, control path and data path per CPU.
+    3) Download trace agent and compile it.
+    4) Enable ftrace in the guest.
+    5) Run trace agent in the guest.
+    6) Open FIFO in a host.
+    7) Start to read trace data by ordering from a host
+    8) Stop to read trace data by ordering from a host
+    9) repeat 7) and 8) with different CPU
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def add_serial_device(params, device, bus, nr, chardev, _id, name):
+        params[
+            "extra_params"
+        ] += " -device {},bus={},nr={},chardev={},id={},name={} ".format(
+            device, bus, nr, chardev, _id, name
+        )
+
+    def add_chardev(params, backend, _id, path):
+        params["extra_params"] += " -chardev {},id={},path={} ".format(
+            backend, _id, path
+        )
+
+    def execute_host_cmd(host_cmd, timeout=60):
+        """
+        Execute the host_cmd on host, limited in timeout period
+
+        param host_cmd: the host_cmd to run on host
+        param timeout: the timeout for running this command
+        return: the output of the host_cmd
+        """
+        logging.info("Executing host command: %s" % host_cmd)
+        cmd_result = process.getstatusoutput(host_cmd, timeout=timeout, shell=True)
+        return cmd_result
+
+    timeout = float(params.get("login_timeout", 240))
+    nums_cpu = int(params.get("smp", 1))
+    v_path = "/tmp/virtio-trace/"
+    execute_host_cmd("mkdir {}".format(v_path))
+    for t in ["in", "out"]:
+        execute_host_cmd("mkfifo /tmp/virtio-trace/agent-ctl-path.{}".format(t))
+        for x in range(int(nums_cpu)):
+            execute_host_cmd("mkfifo {}trace-path-cpu{}.{}".format(v_path, x, t))
```

Comment:
For this case, `chardev_backend == pipe`, according to https://github.com/avocado-framework/avocado-vt/pull/2400
framework will create pipe file before boot vm.

Generate the `path` name for every serial, then call `serials_define_by_params`.